### PR TITLE
feat: add OTLP tracing proxy

### LINKED ISSUE #61: Add OTLP gRPC tracing proxy (localhost:4317)
## Request

Add an OTLP gRPC tracing proxy to `agynd-cli` that runs on `localhost:4317`, injects `agyn.thread.id` as a resource attribute, and forwards spans to the Tracing service.

**Parent plan:** agynio/architecture#120 (Phase 2)

### Overview

New package `internal/tracingproxy/` implements a pass-through OTLP gRPC proxy:
1. Listens on `localhost:4317` (standard OTLP gRPC port)
2. Implements `TraceService/Export` (standard OTLP Collector interface)
3. Injects `agyn.thread.id` resource attribute from `THREAD_ID` env var
4. Forwards enriched request to upstream Tracing service

### New Config Fields (`internal/config/config.go`)

| Field | Env Var | Default | Description |
|---|---|---|---|
| `TracingAddress` | `TRACING_ADDRESS` | `tracing.ziti:443` | Upstream tracing service address |
| `ThreadID` | `THREAD_ID` | (empty) | Thread UUID for attribute injection |

### Proxy Package (`internal/tracingproxy/proxy.go`)

**`Config`** struct:
- `TracingAddress string` — upstream address
- `ThreadID string` — thread ID to inject (empty = skip injection)

**`Proxy`** struct:
- Embeds `UnimplementedTraceServiceServer`
- Fields: `server *grpc.Server`, `upstream TraceServiceClient`, `conn *grpc.ClientConn`, `threadID string`

**`Start(ctx, cfg) → (*Proxy, error)`**:
- Dials upstream with `grpc.NewClient()` + insecure credentials
- Creates gRPC server, registers TraceService
- Listens on `localhost:4317`
- Serves in goroutine

**`Export(ctx, req) → (resp, error)`**:
- If `threadID != ""`: inject `agyn.thread.id` on all `ResourceSpans`
- Forward to upstream

**`Close()`**:
- `GracefulStop()` with 5s timeout fallback to `Stop()`
- Close upstream connection

**`injectThreadID(req, threadID)`**:
- For each `ResourceSpans`: create `Resource` if nil, upsert `agyn.thread.id` attribute

### Daemon Integration

**`internal/daemon/daemon.go`**:
- Add `tracingProxy *tracingproxy.Proxy` field to `Daemon`
- Update `Close()` to call `d.tracingProxy.Close()` before closing gateway conn

**`internal/daemon/agn.go` — `newAgnDaemon()`**:
- Start proxy after `waitForMCPServers()`, before `agnsdk.Start()`
- Add `OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317` to agn env
- Store proxy on Daemon; clean up on failure

**`internal/daemon/daemon.go` — `newCodexDaemon()`**:
- Same pattern: start proxy before `codex.NewClient()`
- Add `OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317` to codex env map

### Dependencies

Add to `go.mod`: `go.opentelemetry.io/proto/otlp v1.10.0`

Provides:
- `go.opentelemetry.io/proto/otlp/collector/trace/v1` — TraceServiceServer/Client
- `go.opentelemetry.io/proto/otlp/common/v1` — KeyValue, AnyValue
- `go.opentelemetry.io/proto/otlp/resource/v1` — Resource
- `go.opentelemetry.io/proto/otlp/trace/v1` — ResourceSpans

### Unit Tests (`internal/tracingproxy/proxy_test.go`)

1. `TestInjectThreadID` — adds attribute to ResourceSpans with existing attributes
2. `TestInjectThreadIDOverwritesExisting` — overwrites existing `agyn.thread.id`
3. `TestInjectThreadIDNilResource` — creates Resource when nil
4. `TestInjectThreadIDMultipleResourceSpans` — injects on all ResourceSpans
5. `TestProxyForwardsToUpstream` — full proxy round-trip with fake upstream, verifies injection + forwarding
6. `TestProxyNoThreadIDPassthrough` — empty ThreadID = no injection, spans still forwarded

### Config Tests

Add tests for `TracingAddress` default, custom override, `ThreadID` parsing, and empty `ThreadID`.

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add OTLP gRPC tracing proxy with thread ID injection and forwarding
- wire proxy into agn/codex daemon startup and env configuration
- add tracing/config tests and update buf generation settings

## Testing
- go test ./...
- go vet ./...

Closes #61

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- go test ./...
  - passed: 7, failed: 0, skipped: 0
- go vet ./...
  - no issues

**Comment by rowan-stein** [comment]
Requesting review. This PR adds an OTLP gRPC tracing proxy to agynd-cli.

**Changes:**
- New `internal/tracingproxy/` package: proxy server on localhost:4317, `agyn.thread.id` injection, upstream forwarding
- Config: `TRACING_ADDRESS` and `THREAD_ID` env vars
- Daemon: proxy lifecycle integration for both agn and codex SDKs
- `OTEL_EXPORTER_OTLP_ENDPOINT` env var added to agent subprocess environments
- Unit tests for injection, overwrite, nil resource, forwarding, passthrough, config parsing

CI is green (build + e2e). Part of [architecture#120](https://github.com/agynio/architecture/issues/120) Phase 2.

File: internal/tracingproxy/proxy.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	return p.upstream.Export(ctx, req)
+}
+
+func (p *Proxy) Close() {
+	if p == nil {
+		return
+	}
+	if p.server != nil {
+		done := make(chan struct{})
+		go func() {
```
Comment (noa-lucent) [inline internal/tracingproxy/proxy.go]: [major] The nil-receiver check (`if p == nil`) and the nil-field guards (`if p.server != nil`, `if p.conn != nil`) are unnecessary defensive fallbacks. `Start` is the only constructor and it always fully initializes both `server` and `conn` before returning a non-nil `*Proxy`. All callers (`Daemon.Close()`) already guard against `tracingProxy` being nil. These checks mask invariant violations instead of surfacing them — if any of these fields were ever nil on a valid `*Proxy`, it would indicate a bug that should fail loudly rather than be silently ignored.

Remove the nil-receiver check and the nil-field guards. If somehow called on an invalid state, let it panic so the root cause is visible.

File: internal/daemon/agn.go [hunk 2]
```diff
@@ -31,25 +32,39 @@ func newAgnDaemon(ctx context.Context, cfg config.Config, version string) (*Daem
...
+	})
+	if err != nil {
+		_ = setup.gatewayConn.Close()
+		return nil, err
+	}
+
 	agnClient, err := agnsdk.Start(ctx, agnsdk.Options{
 		BinaryPath: cfg.AgentBinary,
-		Env:        []string{"AGN_CONFIG_PATH=" + configPath},
+		Env: []string{
```
Comment (noa-lucent) [inline internal/daemon/agn.go:47]: [minor] The OTLP endpoint `"http://localhost:4317"` is a magic string duplicated here and in `daemon.go` (`newCodexDaemon`). The proxy package already defines `listenAddress = "localhost:4317"`. Export a constant (e.g., `tracingproxy.Endpoint`) and derive the full URL from it in both daemon constructors to keep the port/address in a single source of truth.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Good work overall. The proxy implementation is clean, well-tested, and correctly integrated into both daemon paths. The config changes follow existing patterns, and the test coverage is solid with both unit and integration tests.

Two items to address:

1. **[major]** Remove the nil-receiver and nil-field guards in `Proxy.Close()` — they mask invariant violations in internal code. The constructor guarantees fully initialized fields, and callers already guard against nil proxy.

2. **[minor]** The OTLP endpoint string `"http://localhost:4317"` is duplicated across both daemon constructors. Export a constant from the `tracingproxy` package to keep a single source of truth.

**Comment by casey-brooks** [comment]
## Summary
- removed defensive nil checks in tracing proxy Close and exported ListenAddress for OTLP endpoint usage
- updated daemon OTLP endpoint env wiring to derive from tracingproxy.ListenAddress

## Test & Lint Summary
- go test ./...
  - passed: 7, failed: 0, skipped: 0
- go vet ./...
  - no issues

File: internal/tracingproxy/proxy.go (hunk 1 above)
Comment (casey-brooks) [reply internal/tracingproxy/proxy.go]: Updated `Proxy.Close()` to remove nil guards so invalid state panics.

File: internal/daemon/agn.go (hunk 2 above)
Comment (casey-brooks) [reply internal/daemon/agn.go:47]: Exported `tracingproxy.ListenAddress` and now derive the OTLP endpoint in both daemons.

**Review by noa-lucent — APPROVED** [review]
Both requested changes verified in code:

1. `Proxy.Close()` no longer has nil-receiver or nil-field guards — invalid state will now surface immediately.
2. `ListenAddress` is exported and the OTLP endpoint is derived from it in both `newAgnDaemon` and `newCodexDaemon`, eliminating the duplicated magic string.

Clean implementation, good to merge.
